Sample_v3/Samples/SimpleChat/SimpleChatViewController.swift

```diff
@@ -140,33 +144,70 @@ extension DetailViewController: ChannelControllerDelegate {
+extension SimpleChatViewController {
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
         messages.count
     }
     
-    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell: UITableViewCell!
         if let _cell = tableView.dequeueReusableCell(withIdentifier: "MessageCell") {
             cell = _cell
         } else {
             cell = UITableViewCell(style: .default, reuseIdentifier: "MessageCell")
         }
         
+        cell.textLabel?.numberOfLines = 0
+        cell.transform = CGAffineTransform(scaleX: 1, y: -1)
+        
         let message = messages[indexPath.row]
-        cell.textLabel?.text = message.type == .deleted
-            ? "❌ the message was deleted"
-            : "\(message.author.id): \(message.text)"
+        
+        switch message.type {
+        case .deleted:
+            cell?.textLabel?.text = "❌ the message was deleted"
+        case .error:
+            cell?.textLabel?.text = "⚠️ something wrong happened"
+        default:
+            let font = cell.textLabel?.font ?? UIFont.systemFont(ofSize: UIFont.systemFontSize)
+            let boldFont = UIFont(
+                descriptor: font.fontDescriptor.withSymbolicTraits([.traitBold]) ?? font.fontDescriptor,
+                size: font.pointSize
+            )
+            
+            let attributedString = NSMutableAttributedString()
+            attributedString.append(
+                .init(
+                    string: "\(message.author.name ?? message.author.id) ",
+                    attributes: [
+                        NSAttributedString.Key.font: boldFont,
+                        NSAttributedString.Key.foregroundColor: UIColor.forUsername(message.author.id)
+                    ]
+                )
+            )
+            attributedString.append(.init(string: message.text))
+            
+            cell.textLabel?.attributedText = attributedString
+        }
         
         cell.backgroundColor = message.localState == nil ? .white : .lightGray
         
         return cell
     }
     
-    func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+    override func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
         if indexPath.section == tableView.numberOfSections - 1,
             indexPath.row == tableView.numberOfRows(inSection: indexPath.section) - 1 {
             controller?.loadNextMessages()
         }
     }
 }
+
+// MARK: - TableView
+
+extension SimpleChatViewController {
+    func setupTableView() {
+        tableView.contentInsetAdjustmentBehavior = .never
+        tableView.separatorColor = .clear
+        tableView.transform = CGAffineTransform(scaleX: 1, y: -1)
```

Comment:
Is this the preferred way of doing this? Maybe we should rather make the message ordering configurable, no?

Reply:
This is a common way to do it. And the way the ChannelController outputs the updates, I'd have to transform the indices everywhere they're used: (messages.count - index - 1 & indexPath = .init(row: messages.count - indexPath.row - 1 ) and so on.

I think either the ChannelController should have a reversed option or we do it this way to avoid doing index transformation everywhere.

Wdyt?

Reply:
Yeah, we need to expose the ordering option. Let us know next time about if there are strange workarounds you have to do. It's a clear sign we haven't thought about something.

Reply:
I created CIS-314 covering this. We can remove this workaround once that one is implemented.